src/algebra/polynomial/big_operators.lean

```diff
@@ -0,0 +1,95 @@
+/-
+Copyright (c) 2020 Aaron Anderson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Aaron Anderson, Jalex Stark.
+-/
+import algebra.polynomial.basic
+open polynomial finset
+
+/-!
+# Polynomials
+
+Lemmas for the interaction between polynomials and ∑ and ∏.
+
+## Main results
+
+- `nat_degree_prod_eq_of_monic` : the degree of a product of monic polynomials is the product of degrees
+    we prove this only for [comm_semiring R], but it ought to be true for [semiring R] and list.prod
+- `nat_degree_prod_eq` : for polynomials over an integral domain,
+    the degree of the product is the sum of degrees
+- `leading_coeff_prod` : for polynomials over an integral domain,
+    the leading coefficient is the product of leading coefficients
+-/
+
+open_locale big_operators
+
+universes u w
+
+variables {R : Type u} {α : Type w}
+
+namespace polynomial
+
+variable (s : finset α)
+
+section comm_semiring
+variables [comm_semiring R] (f : α → polynomial R)
+
+lemma nat_degree_prod_le : (s.prod f).nat_degree ≤ ∑ i in s, (f i).nat_degree :=
+begin
+  classical,
+  induction s using finset.induction with a s ha hs, { simp },
+  rw [prod_insert ha, sum_insert ha],
+  transitivity (f a).nat_degree + (∏ (x : α) in s, (f x)).nat_degree,
+  apply polynomial.nat_degree_mul_le, linarith,
+end
+
+lemma leading_coeff_prod' (h : ∏ i in s, (f i).leading_coeff ≠ 0) :
```

Comment:
Please announce in a comment that there is a version without `'` below.

Reply:
Done for both `'` lemmas.